pkgs/applications/editors/ldtk/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib, stdenv, fetchurl, unzip, appimage-run }:
+
+stdenv.mkDerivation rec {
+  pname = "ldtk";
+  version = "1.1.3";
+
+  src = fetchurl {
+    url = "https://github.com/deepnight/ldtk/releases/download/v${version}/ubuntu-distribution.zip";
+    sha256 = "sha256-qw7+4k4IH2+9DX4ny8EBbSlyXBrk/y91W04+zWPGupk=";
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  buildInputs = [ appimage-run ];
```

Comment:
There is no a priori prohibition to precompiled packages in NixOS.
Also, it is not always easy to build a package from sources - look at the guy trying to compile KoReader:

https://github.com/NixOS/nixpkgs/pull/178557

I would only suggest to rename it to `ldtk-appimage` in order to differentiate future source-based build expressions.

Reply:
> There is no a priori prohibition to precompiled packages in NixOS.

We always prefer building from source if it is easily possible.

Reply:
> We always prefer building from source if it is easily possible.

Nonetheless it is not a veto or avoidance of precompiled based packaging "at all costs".

> No, we should set meta data to reflect that it isn't build from source.

Sometimes both source-based and precompiled packages are used in Nixpkgs. Obviously they can't have the same exact name in `all-packages.nix`.

Reply:
> Nonetheless it is not a veto or avoidance of precompiled based packaging "at all costs".

If the package is using a standard build system (make, autotools, cmake, etc), has good nixpkgs support (eg not maven, gradle, javascript) and is not proprietary, then fetching a precompiled binary is not acceptable.


> Sometimes both source-based and precompiled packages are used in Nixpkgs. Obviously they can't have the same exact name in `all-packages.nix`.

Yes but that is not the case here. All packages that only have one variant don't get a suffix. Also we should only maintain two versions if the compiled version has major drawbacks.

Reply:
> #178557

I think we can make an exception there because building from source is not easy and upstream has seemingly no good support for linking against system libraries. Going the proposed way there leaves us with a lot of code to maintain which is not worth it.

Reply:
> If the package is using a standard build system (make, autotools, cmake, etc), has good nixpkgs support (eg not maven, gradle, javascript) and is not proprietary, then fetching a precompiled binary is not acceptable. Some packages are really cumbersome to build where we can make exceptions on a case by case basis.

That being said, it requires Haxe, Electron and NPM. Hard but doable.

Reply:
I'd rather have this built from source, too. I don't think I can accomplish it with my current abilities & time, though.